src/wrappers/python/PyOpenEXR.cpp

```diff
@@ -0,0 +1,2755 @@
+    frameBuffer.insert (slice_name,
+                        DeepSlice (_type,
+                                   static_cast<char*>(base_ptr),
+                                   xStride,
+                                   yStride,
+                                   sampleStride,
+                                   xSampling,
+                                   ySampling));
+}
+
+void
+PyPart::writeDeepPixels(MultiPartOutputFile& outfile, const Box2i& dw) const
+{
+    size_t width  = dw.max.x - dw.min.x + 1;
+    size_t height = dw.max.y - dw.min.y + 1;
+
+    auto dw_offset = dw.min.y * width + dw.min.x;
+
+    DeepFrameBuffer frameBuffer;
+        
+    Array2D<unsigned int> sampleCount (height, width);
+    for (size_t y=0; y<height; y++)
+        for (size_t x=0; x<width; x++)
+            sampleCount[y][x] = 0;
+
+    frameBuffer.insertSampleCountSlice (Slice (UINT,
+                                               (char*) (&sampleCount[0][0] - dw_offset),
+                                               sizeof (unsigned int) * 1,       // xStride
+                                               sizeof (unsigned int) * width)); // yStride
+
+    std::vector<std::shared_ptr<Array2DVoidPtr>> sliceDatas;
+
+    for (auto c : channels)
+    {
+        const PyChannel& C = c.second.cast<const PyChannel&>();
+
+        if (C.pixels.dtype().kind() != 'O')
+            throw std::runtime_error("Expected deep pixel array with dtype 'O'");
+
+        C._type = NUM_PIXELTYPES;
+        
+        int nrgba = get_deep_nrgba(C.pixels);
+        if (nrgba == 0)
+            C.insertDeepSlice(frameBuffer, C.name, height, width, nrgba, dw_offset, 0, sampleCount, sliceDatas);
+        else
+        {
+            std::string name_prefix = "";
+            if (C.name != "RGB" && C.name != "RGBA")
+                name_prefix = C.name + ".";
+
+            C.insertDeepSlice(frameBuffer, name_prefix+"R", height, width, nrgba, dw_offset, 0, sampleCount, sliceDatas);
+            C.insertDeepSlice(frameBuffer, name_prefix+"G", height, width, nrgba, dw_offset, 1, sampleCount, sliceDatas);
+            C.insertDeepSlice(frameBuffer, name_prefix+"B", height, width, nrgba, dw_offset, 2, sampleCount, sliceDatas);
+            if (nrgba == 4)
+                C.insertDeepSlice(frameBuffer, name_prefix+"A", height, width, nrgba, dw_offset, 3, sampleCount, sliceDatas);
+        }
+    }
+
+    if (type() == EXR_STORAGE_DEEP_SCANLINE)
+    {
+        DeepScanLineOutputPart part(outfile, part_index);
+        part.setFrameBuffer (frameBuffer);
+        part.writePixels (height);
+    }
+    else 
+    {
+        DeepTiledOutputPart part(outfile, part_index);
+        part.setFrameBuffer (frameBuffer);
+
+        for (int y = 0; y < part.numYTiles (0); y++)
+            for (int x = 0; x < part.numXTiles (0); x++)
+                part.writeTile (x, y, 0);
+    }
+}
+
+//
+// Return whether "name" corresponds to one of the 'R', 'G', 'B', or 'A'
```

Comment:
This sounds like it return a book or 0/1, but in fact it is a mapping between the channel name and array indices, i think.
It might be better to rename it to something along the line of "map to channel indices" or "get channel index" etc...?

Reply:
I renamed it channelNameToRGBA(). Still confusing, but hopefully a little less so.